charm/kafka/reactive/kafka.py

```diff
@@ -263,17 +243,14 @@ def send_data():
 @when_file_changed(os.path.join(KAFKA_SNAP_DATA, "kafka.server.jks"))
 @when_file_changed(os.path.join(KAFKA_SNAP_DATA, "kafka.client.jks"))
 def restart_when_keystore_changed():
-    Kafka().restart()
+    remove_state('kafka.configured')
+    set_flag('kafka.force-reconfigure')
 
 
 @when('tls_client.certs.changed')
 def import_srv_crt_to_keystore():
     for cert_type in ('server', 'client'):
-        keystore_path = os.path.join(
```

Comment:
All the tls handling code should be moved to a separate file, maybe reactive/tls.py?